### PR TITLE
Configure builds for preview and deployment

### LINKED ISSUE #28: Config builds
We should set this project up to produce a production build for deployments:

- Setup [`DefinePlugin`](https://webpack.js.org/plugins/define-plugin/) or [`EnvironmentPlugin`](https://webpack.js.org/plugins/environment-plugin/) to get `NODE_ENV=production` into the bundle, etc
- Config [Netflify for CD](https://www.netlify.com/docs/continuous-deployment/) + our scripts and/or webpack config to build correctly for a given "context".  [`env-universal`](https://www.npmjs.com/package/env-universal) could be helpful here

### PR DESCRIPTION
**Pull request by nason**
Closes #28 

Couple o` extras:
- code splits map component to reduce entry size ~~Causing some strange bug, let's circle back on this one~~ 
- sets up babel cache for development to speed up rebuilds
- adds hashes to output filenames for long term caching support
- bump deps

### CODE REVIEW

**Comment by nason** [comment]
Check that PR status! https://deploy-preview-30--careday.netlify.com/

I tried very quickly to code-split mapbox out of our bundle:

screen shot 2017-04-22 at 3 33 27 pm

But this led to errors. Opened #31 to track.

Still, our deployed bundle is now 236kb (gzipped) vs 1.4MB

File: netlify.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Global settings applied to the whole site.
```
Comment (mattjstar) [inline netlify.toml]: Can you add a comment with this link? https://www.netlify.com/blog/2016/08/30/introducing-deploy-contexts-in-netlify/

File: webpack.client.babel.js [hunk 2]
```diff
@@ -67,12 +77,18 @@ export default function(env) {
       ],
     },
     plugins: [
-      new webpack.EnvironmentPlugin(['CAREDAY_API_URL', 'MAPBOX_TOKEN']),
-      new ExtractTextPlugin('styles.css'),
+      new webpack.LoaderOptionsPlugin({
```
Comment (mattjstar) [inline webpack.client.babel.js]: Why do we need this? 

Looking [here](https://webpack.js.org/plugins/loader-options-plugin/), I was particularly concerned about "In the future this plugin may be removed."

File: netlify.toml (hunk 1 above)
Comment (nason) [inline netlify.toml]: Certainly. Docs around this config are a little lacking, unfortunately. My main reference was https://www.netlify.com/docs/continuous-deployment/

File: webpack.client.babel.js (hunk 2 above)
Comment (nason) [inline webpack.client.babel.js]: webpack v1 had top level `debug` and `production` config properties that affected loader behavior. webpack v2 removed those, and this is a transitional step to maintain dev vs prod behavior in loaders based.

Before they can remove this plugin, loaders will have to make API changes. I wouldn't be terribly concerned about it -- there will be a path forward when the time comes

**Comment by nason** [comment]
I was able to code split mapbox after finding some more information @ https://github.com/mapbox/mapbox-gl-js/issues/4359 and updating our `noParse` rules.

screen shot 2017-04-23 at 12 36 05 pm

**Pull request merged**